actions/views/threads.ts

```diff
@@ -34,3 +44,17 @@ export function manuallyMarkThreadAsUnread(threadId: string, lastViewedAt: numbe
         },
     ]);
 }
+
+export function switchToGlobalThreads() {
+    return (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const state = getState() as GlobalState;
+        const teamUrl = getCurrentRelativeTeamUrl(state);
+
+        if (getIsRhsOpen(state)) {
+            dispatch(suppressRHS);
```

Comment:
I would like to move all the `suppressRHS` dispatch, from individual actions to the `global_threads`/`componentDidMount` lifecycle method, so we don't have to care about it at all.

One thing though, it will introduce a slight delay on closing the RHS while navigating to global threads,
is that OK from a UX point of view @matthewbirtch?

Reply:
@koox00 I guess that would depend on how much of a delay we're talking about. How long are we talking?

Reply:
just after the global threads are shown, now it hides it just before.

Reply:
@koox00 This is probably ok. Can I test it out on this PR?

Reply:
@matthewbirtch @koox00 is that change on the test server though? 

We should be careful about anything that delays opening "threads", even minor delays can have a noticeable impact on usability (it's why we invested in pre-loading unread channels to minimize the short loading time on channel content when executing a channel switch). We already have a significant delay on opening threads on daily (probably because of the non-virtualized list)

Reply:
Sorry... it is not, I didn't had the time today, will do tomorrow morning

Reply:
ok @koox00, let us know when this is in so we can see how it feels

Reply:
@esethna @matthewbirtch the change is on the test server :)

Reply:
@koox00 looks good to me, it doesn't actually delay opening threads it just delays closing the RHS, so my concerns above aren't valid. Thanks!